app/src/main/java/ch/epfl/sdp/blindly/recording/AudioLibraryAdapter.kt

```diff
@@ -214,18 +221,30 @@ class AudioLibraryAdapter(
 
     /**
      * Saves the recording at a given position in the list. It is saved in the app's directory
-     * and can be easily retrieved.
+     * as well as in Firebase storage, and can be easily retrieved.
      *
      * @param position the position of the file we want to save
      */
     private fun saveRecording(position: Int) {
         val filePath = recordList[position].filePath
         val newName = PRESENTATION_AUDIO_NAME
         val currentRecording = File(filePath)
+        val newFile = File("${context.filesDir.absolutePath}/$newName")
         currentRecording.copyTo(
-            File("${context.filesDir.absolutePath}/$newName"),
+            newFile,
             overwrite = true
         )
+        val userId = user.getUserId()
+        recordingPath = "Recordings/$userId-$newName"
+        val storageRef = storage.reference.child(recordingPath)
+        userBuilder.setRecordingPath(recordingPath)
+        storageRef.putFile(Uri.fromFile(newFile)).addOnSuccessListener {
+            startProfileFinished()
+        }.addOnFailureListener {
+            Toast.makeText(context, "Failed to upload the recording. Try again.", Toast.LENGTH_LONG)
+                .show()
+            startProfileFinished()
```

Comment:
Is the profileFinish start not better in the on click listener (as befor line 167), I don't really get the point of putting it in the saveRecording fun

Reply:
because you need to wait for the download to be finished before going to the other page ^^